### ISSUE TITLE
issues with training

### ISSUE THREAD

**Issue by fei1998** [liuqk3/PUT#2]
Thanks for sharing your great work!
During training, i meet the issues like following figures. The loss remains 0 and the reconstruction image is wrong. How to solve them? Thanks! Best wishes!
![problem1](https://user-images.githubusercontent.com/72927722/179359413-8218f6b0-3e6c-48bf-994b-c6b52943bed4.JPG)
![problem2](https://user-images.githubusercontent.com/72927722/179359419-ed5a4e10-74c2-4a34-9997-677587980871.JPG)

**Comment by liuqk3** [comment]
Hi @fei1998 ,

Have you trained P-VAVAE on your dataset? The reconstruction of input data is something wrong. Maybe you can provide your training logs of P-VQVAE.

**Comment by fei1998** [comment]
Thank you very much! The P-VQVAE on my dataset is as shown in the following figures.
![problemb](https://user-images.githubusercontent.com/72927722/179764685-adb9e6b3-3958-443a-a514-8adbf7eddafd.JPG)
![problema](https://user-images.githubusercontent.com/72927722/179764725-86a38b4e-c1c1-45a6-b883-4b80c508d777.JPG)
Best wishes!

**Comment by fei1998** [comment]
Thanks a lot! P-VQVAE on my dataset is shown as the following figures. It seems to be ok?
![problemaa](https://user-images.githubusercontent.com/72927722/179877348-af1582ee-1df1-47d5-9f9c-5ec5ae33361c.JPG)
![problembb](https://user-images.githubusercontent.com/72927722/179877369-f70c992a-38cf-4978-adee-056c0f1a61d6.JPG)
Best wishes!

**Comment by fei1998** [comment]
Thanks a lot! P-VQVAE on my dataset is shown as the following figures. It seems to be wrong?
![problemaa](https://user-images.githubusercontent.com/72927722/179928022-ae89a48f-5e99-4bca-bc41-de12f3e1fc3d.JPG)
![problembb](https://user-images.githubusercontent.com/72927722/179928043-dfbade36-b76d-4c3e-a754-0275c4980106.JPG)
Best wishes!

**Comment by fei1998** [comment]
Thanks a lot! P-VQVAE on my dataset is shown as the following figures. It seems to be wrong?
![problemaa](https://user-images.githubusercontent.com/72927722/179929407-a11f16e7-da84-4782-b800-b0e9e4e8a967.JPG)
![problembb](https://user-images.githubusercontent.com/72927722/179929455-30c463ae-5fdd-43d9-a778-01532d92cbac.JPG)
Best wishes!

**Comment by liuqk3** [comment]
Hi @fei1998 ,

The curves are some different with mines. According to my experience, the reference image should be mostly masked. It is because the reference branch in the decoder is very easy to be learned. However, the codebooks, the encoder and the left part in the decoder are much harder to be learned.  So my suggestion is that increase the mask ratio of the training data and retrain P-VQVAE. 

By the way, the reference branch is infact to increase the overall quality of the model. You can even remove it. If you do so, the output image is totally reconstructed with encoded quantize tokens.  This is very helpful to check if current configuration of P-VQVAE suitable for your dataset.

Best wishes.

**Comment by fei1998** [comment]
Hi, 
Thanks a lot! Let me try again.
Would you like to provide the pretrained P-VQVAE model on ImageNet? Maybe I can finetune it on my dataset.
Best wishes.

**Comment by liuqk3** [comment]
Hi @fei1998 , the pretrained models are all provided since P-VQVAE is a sub-module of PUT. You can get it from our provided model with `model.content_codec`.
